workflow/controller/operator.go

```diff
@@ -4050,6 +4050,19 @@ func setWfPodNamesAnnotation(wf *wfv1.Workflow) {
 	wf.Annotations[common.AnnotationKeyPodNameVersion] = podNameVersion.String()
 }
 
+func setWfTemplateLabel(wf *wfv1.Workflow) {
+	if wf.Spec.WorkflowTemplateRef != nil {
+		if wf.ObjectMeta.Labels == nil {
+			wf.ObjectMeta.Labels = map[string]string{}
+		}
+		if wf.Spec.WorkflowTemplateRef.ClusterScope {
+			wf.ObjectMeta.Labels[common.LabelKeyClusterWorkflowTemplate] = wf.Spec.WorkflowTemplateRef.Name
+		} else {
+			wf.ObjectMeta.Labels[common.LabelKeyWorkflowTemplate] = wf.Spec.WorkflowTemplateRef.Name
+		}
+	}
```

Comment:
let's invert this for less nesting
```suggestion
	if wf.Spec.WorkflowTemplateRef == nil {
		return
	}
	if wf.ObjectMeta.Labels == nil {
		wf.ObjectMeta.Labels = map[string]string{}
	}
	
	if wf.Spec.WorkflowTemplateRef.ClusterScope {
		wf.ObjectMeta.Labels[common.LabelKeyClusterWorkflowTemplate] = wf.Spec.WorkflowTemplateRef.Name
	} else {
		wf.ObjectMeta.Labels[common.LabelKeyWorkflowTemplate] = wf.Spec.WorkflowTemplateRef.Name
	}
```